contrib/completion/fish/docker.fish

```diff
@@ -117,448 +133,578 @@ complete -c docker -f -n '__fish_docker_no_subcommand' -s v -l version -d 'Print
-complete -c docker -A -f -n '__fish_seen_subcommand_from pause' -a '(__fish_print_docker_containers running)' -d "Container"
+complete -c docker -f -n '__fish_docker_seen_subcommand_one_level_deep pause' -a '(__fish_print_docker_containers running)' -d "Container"
 
 # ps
 complete -c docker -f -n '__fish_docker_no_subcommand' -a ps -d 'List containers'
-complete -c docker -A -f -n '__fish_seen_subcommand_from ps' -s a -l all -d 'Show all containers. Only running containers are shown by default.'
-complete -c docker -A -f -n '__fish_seen_subcommand_from ps' -l before -d 'Show only container created before Id or Name, include non-running ones.'
-complete -c docker -A -f -n '__fish_seen_subcommand_from ps' -s f -l filter -d 'Provide filter values. Valid filters:'
-complete -c docker -A -f -n '__fish_seen_subcommand_from ps' -l help -d 'Print usage'
-complete -c docker -A -f -n '__fish_seen_subcommand_from ps' -s l -l latest -d 'Show only the latest created container, include non-running ones.'
-complete -c docker -A -f -n '__fish_seen_subcommand_from ps' -s n -d 'Show n last created containers, include non-running ones.'
-complete -c docker -A -f -n '__fish_seen_subcommand_from ps' -l no-trunc -d "Don't truncate output"
-complete -c docker -A -f -n '__fish_seen_subcommand_from ps' -s q -l quiet -d 'Only display numeric IDs'
-complete -c docker -A -f -n '__fish_seen_subcommand_from ps' -s s -l size -d 'Display total file sizes'
-complete -c docker -A -f -n '__fish_seen_subcommand_from ps' -l since -d 'Show only containers created since Id or Name, include non-running ones.'
+complete -c docker -f -n '__fish_docker_seen_subcommand_one_level_deep ps' -s a -l all -d 'Show all containers. Only running containers are shown by default.'
+complete -c docker -f -n '__fish_docker_seen_subcommand_one_level_deep ps' -l before -d 'Show only container created before Id or Name, include non-running ones.'
+complete -c docker -f -n '__fish_docker_seen_subcommand_one_level_deep ps' -s f -l filter -d 'Provide filter values. Valid filters:'
+complete -c docker -f -n '__fish_docker_seen_subcommand_one_level_deep ps' -l help -d 'Print usage'
+complete -c docker -f -n '__fish_docker_seen_subcommand_one_level_deep ps' -s l -l latest -d 'Show only the latest created container, include non-running ones.'
+complete -c docker -f -n '__fish_docker_seen_subcommand_one_level_deep ps' -s n -d 'Show n last created containers, include non-running ones.'
+complete -c docker -f -n '__fish_docker_seen_subcommand_one_level_deep ps' -l no-trunc -d "Don't truncate output"
+complete -c docker -f -n '__fish_docker_seen_subcommand_one_level_deep ps' -s q -l quiet -d 'Only display numeric IDs'
+complete -c docker -f -n '__fish_docker_seen_subcommand_one_level_deep ps' -s s -l size -d 'Display total file sizes'
+complete -c docker -f -n '__fish_docker_seen_subcommand_one_level_deep ps' -l since -d 'Show only containers created since Id or Name, include non-running ones.'
 
 # pull
 comple
```

Comment:
Hm, I'd have to dive a bit deeper into this, but it still looks to be not fully working, for example, this completion doesn't seem to trigger (it defaults to completing local filenames)


I also noticed that there's no completion for the "old" `docker volumes` command.

I wonder if we can come with a utility function that accepts the exact list of subcommands, e.g.;

```
__fish_docker_seen_subcommands volume rm
```

Did a quick attempt, but it doesn't fully work yet (i.e. `docker volume inspect` would still complete with `docker volume inspect inspect` (or other `docker volume` subcommands);

```
function __fish_docker_seen_subcommands --description 'Detect a subcommand'
    set -l cmd (commandline -poc)
    set -e cmd[1]
    if [ (count $cmd) -lt (count $argv) ]
      return 1
    end
    for i in (seq (count $argv))
      if [ $cmd[$i] != $argv[$i] ]
        return 1
      end
    end
    return 0
end

complete -c docker -f -n '__fish_docker_no_subcommand' -a hello
complete -c docker -f -n '__fish_docker_seen_subcommands hello' -a world
complete -c docker -f -n '__fish_docker_seen_subcommands hello world' -a '(__fish_print_docker_volume_names)'
```